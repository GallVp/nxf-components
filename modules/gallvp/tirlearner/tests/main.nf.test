nextflow_process {

    name "Test Process TIRLEARNER"
    script "../main.nf"
    process "TIRLEARNER"

    tag "modules"
    tag "modules_gallvp"
    tag "tirlearner"


    test("actinidia_chinensis-genome_1_fasta_gz-success") {

        when {
            process {
                """
                input[0] = [
                    [ id: 'test' ],
                    file(params.modules_testdata_base_path + 'genomics/eukaryotes/actinidia_chinensis/genome/chr1/genome.fasta.gz', checkIfExists: true)
                ]
                input[1] = 'others'
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert file(process.out.log[0][1]).text.contains('Module 4 Begin') },
                { assert snapshot(
                    process.out.fasta,
                    process.out.gff,
                    process.out.filtered_fasta,
                    process.out.filtered_gff,
                    process.out.versions
                    ).match()
                }
            )
        }

    }

    test("sarscov2-genome") {

        when {
            process {
                """
                input[0] = [
                    [ id: 'test' ],
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.fasta', checkIfExists: true)
                ]
                input[1] = 'others'
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert file(process.out.log[0][1]).text.contains('ValueError: All objects passed were None') },
                { assert snapshot(process.out.versions).match() }
            )
        }

    }

    test("stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id: 'test' ],
                    file(params.modules_testdata_base_path + 'genomics/eukaryotes/actinidia_chinensis/genome/chr1/genome.fasta.gz', checkIfExists: true)
                ]
                input[1] = 'others'
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }
}
