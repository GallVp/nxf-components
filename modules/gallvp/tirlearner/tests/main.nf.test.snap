{
    "sarscov2-genome": {
        "content": [
            [
                "versions.yml:md5,054eb77998e320f8b70cae832f7bff59"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-12-17T15:23:12.222704"
    },
    "actinidia_chinensis-genome_1_fasta_gz-success": {
        "content": [
            [
                [
                    {
                        "id": "test"
                    },
                    "test.fa:md5,65987decb67d1e3f95b993f86d14c0d0"
                ]
            ],
            [
                [
                    {
                        "id": "test"
                    },
                    "test.gff3:md5,90bf040e7b29763351237207297dc801"
                ]
            ],
            [
                [
                    {
                        "id": "test"
                    },
                    "test.filtered.fa:md5,65987decb67d1e3f95b993f86d14c0d0"
                ]
            ],
            [
                [
                    {
                        "id": "test"
                    },
                    "test.filtered.gff3:md5,90bf040e7b29763351237207297dc801"
                ]
            ],
            [
                "versions.yml:md5,054eb77998e320f8b70cae832f7bff59"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-12-17T15:22:50.978766"
    },
    "stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.log:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test.fa:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test"
                        },
                        "test.gff3:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "test"
                        },
                        "test.filtered.fa:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "4": [
                    [
                        {
                            "id": "test"
                        },
                        "test.filtered.gff3:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "5": [
                    "versions.yml:md5,054eb77998e320f8b70cae832f7bff59"
                ],
                "fasta": [
                    [
                        {
                            "id": "test"
                        },
                        "test.fa:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "filtered_fasta": [
                    [
                        {
                            "id": "test"
                        },
                        "test.filtered.fa:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "filtered_gff": [
                    [
                        {
                            "id": "test"
                        },
                        "test.filtered.gff3:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "gff": [
                    [
                        {
                            "id": "test"
                        },
                        "test.gff3:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "log": [
                    [
                        {
                            "id": "test"
                        },
                        "test.log:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,054eb77998e320f8b70cae832f7bff59"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-12-17T15:23:25.302007"
    }
}